### DISCUSSION TITLE
keepPreviousData is cool - but is it the right UI pattern?

### DISCUSSION THREAD

**Discussion by cmacdonnacha** [mswjs/msw#702] [Q&A] [answered]
I just used `keepPreviousData` from [here](https://react-query.tanstack.com/guides/paginated-queries) and it's really cool and all but wouldn't the user still want to know that something is happening while they wait for new data (the next page) to arrive? Just imagine a user on a slow network, they click "Next Page" but nothing is visibly happening because the old data is still displayed, so they click "Next page" a couple of more times in frustration. Whereas if they were given some feedback like a "Loading" spinner at least they'd know they have to wait. 

Am I missing something?

**Comment by kettanaito** [comment, marked as answer]
Hi, @cmacdonnacha. I'm afraid you're reached out to the wrong tool. While MSW allows mocking any responses, even to the requests made with `react-query`, it has no control over how you use your request clients. This looks like a question to raise in the `react-query` discussions board.

**Reply by cmacdonnacha** [reply]
oops so sorry, wrong repo!

I would really appreciate your help on this question though: https://github.com/mswjs/msw/discussions/698
